### PR TITLE
GF-41 - Redux implementation for projects

### PR DESCRIPTION
**Pull request by iamdavidpetric**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,12 +1,17 @@
 import React from 'react';
 import ReactDOM from 'react-dom/client';
 
+import store from './process/redux/store';
```
Comment (iamalextodoran) [inline src/index.js]: Length

**Review by MIKALAKE — APPROVED** [review]

File: src/process/redux/projectSlice.js [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+        `http://localhost:3000/v1/projects/${payload.id}`
+      );
+      return response.data;
+    } catch (err) {
+      return err.message;
+    }
+  }
+);
+
+export const getProjectAsync = createAsyncThunk(
```
Comment (iamalextodoran) [inline src/process/redux/projectSlice.js]: This could be renamed to getMyProjects

File: src/process/redux/projectSlice.js [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+        formData.append('team_members_attributes[][name]', member.name);
+        formData.append(
+          'team_members_attributes[][avatar_url]',
+          member.avatar_url
+        );
+      }
+    });
+    formData.append('category', project.category);
+    formData.append('statement', project.statement);
+    formData.append('user_id', project.user_id);
```
Comment (iamalextodoran) [inline src/process/redux/projectSlice.js:88]: Can you actually create new projects?

File: src/process/redux/projectSlice.js [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+);
+
+const projectSlice = createSlice({
+  name: 'projects',
+  initialState: {
+    featuredProject: {},
+    mostInvested: [],
+    projects: [],
+    selectedProject: {},
+    quickInfo: {}
```
Comment (iamalextodoran) [inline src/process/redux/projectSlice.js]: abc those.

File: src/process/redux/projectSlice.js [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+  },
+
+  reducers: {
+    deleteProject: (state, action) => {
+      return state.filter(project => project.id !== action.payload.id);
+    },
```
Comment (iamalextodoran) [inline src/process/redux/projectSlice.js]: Not used.

File: src/process/redux/projectSlice.js [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+
+    [editProjectAsync.fulfilled]: (state, { payload }) => {
+      const newProjects = [...state.projects];
+      const index = project => project.id === payload.id;
+      const selectedProject = newProjects.findIndex(index);
+      newProjects[selectedProject] = payload;
+      return {
+        ...state,
+        projects: newProjects
+      };
```
Comment (iamalextodoran) [inline src/process/redux/projectSlice.js]: Does this not fit in one row?

File: src/process/redux/projectSlice.js [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+    },
+
+    [deleteProjectAsync.fulfilled]: (state, { payload }) => {
+      const filter = project => project.id !== payload.id;
+      const filteredProjects = state.projects.filter(filter);
+      return {
+        ...state,
+        projects: filteredProjects
```
Comment (iamalextodoran) [inline src/process/redux/projectSlice.js]: One line?

File: src/show/pages/private/CreateProject/Subviews/DeadlineStep/index.js [hunk 8]
```diff
@@ -17,10 +17,13 @@ const DeadlineStep = ({ nextStep, previousStep, project, setProject }) => {
...
               className='flex justify-center text-center w-full '
-              selected={project?.deadline}
-              onChange={deadline =>
-                setProject({ ...project, deadline: deadline })
+              selected={
+                project.deadline ? new Date(project.deadline) : new Date()
               }
+              onChange={deadline => {
+                setProject({ ...project, deadline: deadline });
+                console.log(deadline);
```
Comment (iamalextodoran) [inline src/show/pages/private/CreateProject/Subviews/DeadlineStep/index.js]: Remove console log and make it a one liner again.

File: src/show/pages/private/CreateProject/index.js [hunk 9]
```diff
@@ -1,9 +1,8 @@
-import axios from 'axios';
 import { useState } from 'react';
+import { useDispatch } from 'react-redux';
 import StepWizard from 'react-step-wizard';
-import { useNavigate } from 'react-router-dom';
+import { createProjectAsync } from '../../../../process/redux/projectSlice';
```
Comment (iamalextodoran) [inline src/show/pages/private/CreateProject/index.js]: Should be grouped bellow

File: src/show/pages/private/EditProject/index.js [hunk 10]
```diff
@@ -1,8 +1,14 @@
-import axios from 'axios';
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import StepWizard from 'react-step-wizard';
```
Comment (iamalextodoran) [inline src/show/pages/private/EditProject/index.js:2]: Length

File: src/show/pages/private/EditProject/index.js [hunk 11]
```diff
@@ -1,8 +1,14 @@
...
-import { useEffect, useState } from 'react';
+
 import { useParams } from 'react-router-dom';
```
Comment (iamalextodoran) [inline src/show/pages/private/EditProject/index.js:3]: Should be with those above

File: src/show/pages/private/EditProject/index.js [hunk 12]
```diff
@@ -16,79 +22,41 @@ import {
...
-        <TeamMembersStep project={project} setProject={setProject} />
+        <CategoryStep project={selectedProject} setProject={setProject} />
+        <TitleStep project={selectedProject} setProject={setProject} />
+        <DescriptionStep project={selectedProject} setProject={setProject} />
+        <GoalStep project={selectedProject} setProject={setProject} />
+        <FeeStep project={selectedProject} setProject={setProject} />
+        {/* <DeadlineStep project={selectedProject} setProject={setProject} /> */}
+        <StatementStep project={selectedProject} setProject={setProject} />
+        {/* <ImagesStep project={selectedProject} setProject={setProject} /> */}
+        {/* <TeamMembersStep project={selectedProject} setProject={setProject} /> */}
```
Comment (iamalextodoran) [inline src/show/pages/private/EditProject/index.js]: What's wrong with those? I only allowed commenting the deadline step. Fix those.

File: src/show/pages/private/MyProjects/index.js [hunk 13]
```diff
@@ -1,7 +1,12 @@
-import axios from 'axios';
 import { useNavigate } from 'react-router-dom';
 import { Fragment, useState, useEffect } from 'react';
 import { FaPencilAlt, FaTrash } from 'react-icons/fa';
+import { useDispatch, useSelector } from 'react-redux';
+
+import {
+  deleteProjectAsync,
+  getProjectAsync
+} from '../../../../process/redux/projectSlice';
```
Comment (iamalextodoran) [inline src/show/pages/private/MyProjects/index.js:9]: No empty line after

File: src/show/pages/private/MyProjects/index.js [hunk 14]
```diff
@@ -11,31 +16,22 @@ import { Button, ProgressBar, Modal } from '../../../components';
...
   const navigate = useNavigate();
+  const dispatch = useDispatch();
 
-  useEffect(() => {
-    axios
-      .get(`http://localhost:3000/v1/projects/my_projects`)
-      .then(res => setMyProjects(res.data))
-      .catch(err => err);
-  }, []);
+  const projects = useSelector(state => state.projects.projects);
```
Comment (iamalextodoran) [inline src/show/pages/private/MyProjects/index.js]:
```suggestion
  const { projects } = useSelector(state => state.projects);
```

File: src/show/pages/private/MyProjects/index.js [hunk 15]
```diff
@@ -44,12 +40,13 @@ const MyProjects = () => {
   return (
     <Fragment>
       <div className='h-full w-full'>
-        {myProjects.length > 0 ? (
+        {projects ? (
```
Comment (iamalextodoran) [inline src/show/pages/private/MyProjects/index.js]: Also I really like myProjects....
```suggestion
        {projects.length > 0 ? (
```

File: src/show/pages/public/Home/index.js [hunk 16]
```diff
@@ -1,33 +1,29 @@
...
+import { useDispatch, useSelector } from 'react-redux';
 
 import { Button, Card, ProgressBar } from '../../../components';
 import { PROJECT_PATH } from '../../../../process/routes/paths';
 
-const Home = () => {
-  const [featuredProject, setFeaturedProject] = useState({});
-  const [mostInvested, setMostInvested] = useState([]);
-  const [quickInfo, setQuickInfo] = useState({});
+import {
```
Comment (iamalextodoran) [inline src/show/pages/public/Home/index.js:7]: No Empty line before

**Review by iamalextodoran — CHANGES_REQUESTED** [review]

**Comment by iamalextodoran** [comment]
GF-41 - Redux implementation for projects

File: src/process/redux/projectSlice.js (hunk 3 above)
Comment (iamdavidpetric) [reply src/process/redux/projectSlice.js:88]: yes boss i only tell you the truth

File: src/process/redux/projectSlice.js (hunk 2 above)
Comment (iamdavidpetric) [reply src/process/redux/projectSlice.js]: getMyProjectsAsync

File: src/index.js (hunk 1 above)
Comment (iamdavidpetric) [reply src/index.js]: yes sir

File: src/process/redux/projectSlice.js (hunk 4 above)
Comment (iamdavidpetric) [reply src/process/redux/projectSlice.js]: done

File: src/process/redux/projectSlice.js (hunk 5 above)
Comment (iamdavidpetric) [reply src/process/redux/projectSlice.js]: deleted deleteProject

File: src/process/redux/projectSlice.js (hunk 6 above)
Comment (iamdavidpetric) [reply src/process/redux/projectSlice.js]: it's chunky

File: src/process/redux/projectSlice.js (hunk 7 above)
Comment (iamdavidpetric) [reply src/process/redux/projectSlice.js]: it's chunky

File: src/show/pages/private/CreateProject/Subviews/DeadlineStep/index.js (hunk 8 above)
Comment (iamdavidpetric) [reply src/show/pages/private/CreateProject/Subviews/DeadlineStep/index.js]: --------

File: src/show/pages/private/CreateProject/index.js (hunk 9 above)
Comment (iamdavidpetric) [reply src/show/pages/private/CreateProject/index.js]: sure

File: src/show/pages/private/EditProject/index.js (hunk 11 above)
Comment (iamdavidpetric) [reply src/show/pages/private/EditProject/index.js:3]: allright

File: src/show/pages/private/EditProject/index.js (hunk 12 above)
Comment (iamdavidpetric) [reply src/show/pages/private/EditProject/index.js]: pls don;t shoot me

Comment (iamdavidpetric) [reply src/show/pages/private/EditProject/index.js]: but something is wrong with these 2 steps (i can;t edit them)

File: src/show/pages/private/MyProjects/index.js (hunk 13 above)
Comment (iamdavidpetric) [reply src/show/pages/private/MyProjects/index.js:9]: ok

File: src/show/pages/private/MyProjects/index.js (hunk 14 above)
Comment (iamdavidpetric) [reply src/show/pages/private/MyProjects/index.js]: 👍

File: src/show/pages/private/MyProjects/index.js (hunk 15 above)
Comment (iamdavidpetric) [reply src/show/pages/private/MyProjects/index.js]: biniuca

File: src/show/components/Navbar/index.js [hunk 17]
```diff
@@ -41,6 +38,14 @@ const Navbar = () => {
               label='explore'
             />
           </div>
+          <div onClick={() => navigate(MY_PROJECTS_PATH)}>
+            {' '}
+            <Button
+              variant='secondary'
+              iconLeft={<TbStairsUp className='mx-1' size='1.5rem' />}
+              label='MyProjects'
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js:46]:
```suggestion
              label='My Projects'
```

File: src/show/pages/private/CreateProject/index.js [hunk 18]
```diff
@@ -1,9 +1,9 @@
-import axios from 'axios';
 import { useState } from 'react';
+import { useDispatch } from 'react-redux';
+
 import StepWizard from 'react-step-wizard';
-import { useNavigate } from 'react-router-dom';
+import { createProjectAsync } from '../../../../process/redux/projectSlice';
```
Comment (iamalextodoran) [inline src/show/pages/private/CreateProject/index.js]: Still no bueno. Think again about grouping and ordering

File: src/show/pages/private/EditProject/index.js [hunk 19]
```diff
@@ -1,8 +1,13 @@
...
-import { useEffect, useState } from 'react';
 import { useParams } from 'react-router-dom';
-import { useNavigate } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+
+import {
+  setSelectedProject,
+  editProjectAsync,
+  getProjectProjectAsync
+} from '../../../../process/redux/projectSlice';
```
Comment (iamalextodoran) [inline src/show/pages/private/EditProject/index.js:10]: No empty line after this.

File: src/show/pages/private/MyProjects/index.js [hunk 20]
```diff
@@ -1,41 +1,33 @@
...
 
 import {
-  EDIT_PROJECT_PATH,
-  PROJECT_PATH
-} from '../../../../process/routes/paths';
+  deleteProjectAsync,
+  getMyProjectsAsync
+} from '../../../../process/redux/projectSlice';
+import { EDIT_PROJECT_PATH, PROJECT_PATH } from '../../../../process/routes/paths';
 import { Button, ProgressBar, Modal } from '../../../components';
```
Comment (iamalextodoran) [inline src/show/pages/private/MyProjects/index.js:10]: By length.

File: src/show/pages/private/MyProjects/index.js [hunk 21]
```diff
@@ -109,7 +96,10 @@ const MyProjects = () => {
               <div className='flex items-center justify-between w-full'>
                 <div className='w-full'>
                   <Button
-                    onClick={e => deleteProject(e)}
+                    onClick={e => {
+                      handleDeleteClick(e);
+                      setDeleteProjectModal(false);
```
Comment (iamalextodoran) [inline src/show/pages/private/MyProjects/index.js]: Close the modal in the handledeleteclick function and you can do this in one line after.

**Review by iamalextodoran — CHANGES_REQUESTED** [review]

File: src/show/components/Navbar/index.js (hunk 17 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js:46]: ↔️

File: src/show/pages/private/CreateProject/index.js (hunk 18 above)
Comment (iamdavidpetric) [reply src/show/pages/private/CreateProject/index.js]: 🧼

File: src/show/pages/private/EditProject/index.js (hunk 19 above)
Comment (iamdavidpetric) [reply src/show/pages/private/EditProject/index.js:10]: :empty

File: src/show/pages/private/MyProjects/index.js (hunk 20 above)
Comment (iamdavidpetric) [reply src/show/pages/private/MyProjects/index.js:10]: 🥬

File: src/show/pages/private/MyProjects/index.js (hunk 21 above)
Comment (iamdavidpetric) [reply src/show/pages/private/MyProjects/index.js]: roger that

**Review by iamalextodoran — APPROVED** [review]

**Pull request merged**